### PR TITLE
Add docker CI

### PR DESCRIPTION
**Pull request by EkkoG**
### Background

Make deploy juicity server easier.

### Checklist

- [x] The Pull Request has been fully tested
- [x] There's an entry in the CHANGELOGS
- [ ] There is a user-facing docs PR against https://github.com/juicity/juicity

### Full Changelogs

- Add docker CI

### Issue Reference

no

### Test Result

https://github.com/EkkoG/juicity/pkgs/container/juicity/139716504?tag=v0.3.1
https://github.com/EkkoG/juicity/pkgs/container/juicity/139709638?tag=main

### CODE REVIEW

File: .github/workflows/docker-publish.yml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+on:
+  push:
+    branches: [ "main" ]
+    # Publish semver tags as releases.
+    tags: [ 'v*.*.*' ]
```
Comment (yqlbu) [inline .github/workflows/docker-publish.yml:12]:
```suggestion
    tags: [ 'v*' ]
```

**Review by yqlbu — CHANGES_REQUESTED** [review]
Could we please not pin the `sha` as a workflow reference?

We may pin `tag` as a reference instead.

E.g.

```yaml
# https://github.com/docker/metadata-action/tags
uses: docker/build-push-action@v5.0.0
```

**Comment by EkkoG** [comment]
All issue resolved.

New test result
https://github.com/EkkoG/juicity/actions/runs/6600686251
https://github.com/EkkoG/juicity/actions/runs/6600687992

**Review by yqlbu — APPROVED** [review]

**Comment by yqlbu** [comment]
Thanks for your contribution!

**Comment by sjtuross** [comment]
Great to have the official docker build. Please also add linux/arm64 arch.

**Comment by EkkoG** [comment]
> Great to have the official docker build. Please also add linux/arm64 arch.

@sjtuross https://github.com/juicity/juicity/pull/133

**Pull request merged**